ui/src/app/lib/atlasmap-data-mapper/models/mapping.model.ts

```diff
@@ -343,4 +368,26 @@ export class MappingModel {
     }
   }
 
+  /**
+   * Remove any trailing padding fields for the mapped field array.  This occurs when a user moves
+   * a mapped element above the last padding field.
+   *
+   * @param mappedFields
+   * @param trailing - Remove trailing padding fields only
+   */
+  clearExtraPaddingFields(mappedFields: MappedField[], trailing: boolean): void {
```

Comment:
Is there a reason moving this from MappingManagementService? I prefer having more complex logic than basic CRUD in services rather than model itself.

Reply:
ya - I debated that myself.  The reason it's there is that it's needed by remoceField() in mapping.model.  If I put it into mapping-management.service then I'll introduce a circular dependency between model and mms.

Reply:
So invoke it from here instead?
https://github.com/atlasmap/atlasmap/pull/1181/files#diff-33a65a8e6d54bd6716a9670a71fe3408L306
my vscode says this is the only place invoking MappingModel.removeField()
Then we can keep MappingModel.removeField() from handling other fields (i.e. padding)

Reply:
Makes sense - move the padding clear method back over to the management service.